utils/s3/client.cc

```diff
@@ -112,7 +128,8 @@ shared_ptr<client> client::make(std::string endpoint, endpoint_config_ptr cfg, s
 }
 
 future<> client::authorize(http::request& req) {
-    if (!_credentials || std::chrono::system_clock::now() >= _credentials.expires_at) {
+    if (!_credentials) {
```

Comment:
I'd also kick the background update timer instantly in constructor so that by the time the very first request comes here the creds are already available. But it's an optimization

Reply:
something like 
```
_creds_update_timer.arm(std::chrono::steady_clock::now());
_creds_invalidation_timer.arm(std::chrono::steady_clock::now() + 1h);
```
in the constructor?